src/context/PortfolioProvider/PortfolioContext.tsx

```diff
@@ -171,6 +182,79 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
     })()
   }, [assetsById, chainAdapter, dispatch, wallet])
 
+  const txIds = useSelector(selectTxIds)
+  const txsById = useSelector(selectTxs)
+  const txHistoryStatus = useSelector(selectTxHistoryStatus)
+
+  /**
+   * monitor for new pending txs, add them to a set, so we can monitor when they're confirmed
+   */
+  useEffect(() => {
+    // we only want to refetch portfolio if a new tx comes in after we're finished loading
+    if (txHistoryStatus !== 'loaded') return
+    // don't fire with nothing connected
+    if (isEmpty(accountSpecifiers)) return
+    // grab the most recent txId
+    const txId = head(txIds)!
+    // grab the actual tx
+    const tx = txsById[txId]
+    // always wear protection, or don't it's your choice really
+    if (!tx) return
+
+    if (tx.caip2 === cosmosChainId) {
+      // cosmos txs only come in when they're confirmed, so refetch that account immediately
+      return refetchAccountByTxId(txId)
+    } else {
+      /**
+       * the unchained getAccount call does not include pending txs in the portfolio
+       * add them to a set, and the two effects below monitor the set on pending txs
+       */
+      if (tx.status === 'pending') setPendingTxIds(new Set([...pendingTxIds, txId]))
+    }
+
+    // txsById changes on each tx - as txs have more confirmations
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dispatch, txIds])
```

Comment:
`dispatch` is not needed here anymore

Reply:
Nice catch! We can also ditch it in other `useEffect` calls as well since we have `// eslint-disable-next-line react-hooks/exhaustive-deps`. The only reason `dispatch` is often included as a dep is to satisfy the linter, it should be referentially stable.

Reply:
noice catch, i will remove it here. 

generally, i want to start imposing this pattern - if we disable the effect dependencies, we should force people to describe exactly *why* they were removed.